### PR TITLE
[4.1] [PHP 8.1] Compatibility of htmlentities() types in tinymce DisplayTrait

### PR DESCRIPTION
**Pull request by beat**
Pull Request for Issue # none

### Summary of Changes

This is the Joomla 4.1 identical correspondent pendent of the Joomla 3.10 PR #36804 that got merged, and replaces previous Joomla 4.0 PR #36761 that got outdated due to the cool refactoring of editors into Traits.

In PHP, `htmlentities()` has following parameters types: 2nd is a non-null int, 4th is à non-null bool.
```
 htmlentities(
    string $string,
    int $flags = ENT_QUOTES | ENT_SUBSTITUTE | ENT_HTML401,
    ?string $encoding = null,
    bool $double_encode = true
): string
```

ref.: https://www.php.net/manual/en/function.htmlentities.php

This PR does replace null by their type-conversions: (int) null -> 0, and (bool) null -> false.

**Disclaimer:** I'm only fixing PHP 8.1 compatibility here.

### Testing Instructions

A code-review should be enough here.

I found it with CB internal PMS / New Message, but I guess PHP 8.1 on any tinymce window should show it too.

### Actual result BEFORE applying this Pull Request

This is the result on 4.0 (in 4.1-dev, it's erroring on line 380 of DisplayTrait.php instead of line 538 of tinymce.php):

* Deprecated: htmlentities(): Passing null to **parameter # 2** ($flags) of type int is deprecated in /home/beat/www/4.0/plugins/editors/tinymce/tinymce.php on line 538
* Deprecated: htmlentities(): Passing null to **parameter # 4** ($double_encode) of type bool is deprecated in /home/beat/www/4.0/plugins/editors/tinymce/tinymce.php on line 538

### Expected result AFTER applying this Pull Request

No errors.

### Documentation Changes Required

None.

### CODE REVIEW

File: plugins/editors/tinymce/src/PluginTraits/DisplayTrait.php [hunk 1]
```diff
@@ -377,7 +377,7 @@ public function onDisplay(
 		{
 			$externalPlugins['jdragndrop'] = HTMLHelper::_('script', 'plg_editors_tinymce/plugins/dragdrop/plugin.min.js', ['relative' => true, 'version' => 'auto', 'pathOnly' => true]);
 			$uploadUrl                     = Uri::base(false) . 'index.php?option=com_media&format=json&url=1&task=api.files';
-			$uploadUrl                     = $this->app->isClient('site') ? htmlentities($uploadUrl, null, 'UTF-8', null) : $uploadUrl;
+			$uploadUrl                     = $this->app->isClient('site') ? htmlentities($uploadUrl, 0, 'UTF-8', false) : $uploadUrl;
```
Comment (PhilETaylor) [inline plugins/editors/tinymce/src/PluginTraits/DisplayTrait.php]: Please use the PHP Constant that was designed for this purpose `ENT_NOQUOTES` - see https://3v4l.org/70HcC

```suggestion
			$uploadUrl                     = $this->app->isClient('site') ? htmlentities($uploadUrl, ENT_NOQUOTES, 'UTF-8', false) : $uploadUrl;
```

Comment (beat) [reply plugins/editors/tinymce/src/PluginTraits/DisplayTrait.php]: :+1: Good tip, doesn't change the value of 0 but improves the readability. Merging, Thank you, @PhilETaylor

**Comment by Quy** [comment]
Please fix conflicts.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 5356a798c50c98b4bf33140d29002be249d441c1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36846">issues.joomla.org/tracker/joomla-cms/36846</a>.</sub>

**Comment by laoneo** [comment]
I have tested this item :white_check_mark: successfully on 5356a798c50c98b4bf33140d29002be249d441c1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36846">issues.joomla.org/tracker/joomla-cms/36846</a>.</sub>

**Comment by laoneo** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36846">issues.joomla.org/tracker/joomla-cms/36846</a>.</sub>

**Comment by bembelimen** [comment]
Thx

**Pull request merged**